systems/introduction/vm.go

```diff
@@ -28,16 +30,47 @@ const (
 func compute(memory []byte) {
 
 	registers := [3]byte{8, 0, 0} // PC, R1 and R2
+	var pcincr byte               // Amount to increment the PC each cycle
 
 	// Keep looping, like a physical computer's clock
 	for {
+		pcincr = 3 // Default instruction length is three bytes
+
+		op := memory[registers[0]]
+		arg1 := memory[registers[0]+1]
+		arg2 := memory[registers[0]+2]
```

Comment:
What if you encounter a `Halt` instruction at the last memory location (255)?

Reply:
Good call! I was being lazy 😄